content/Watsonx/WatsonxData/108.md

```diff
@@ -0,0 +1,119 @@
+---
+title: '108: Time Travel and Rollback'
+timeToComplete: 45
+updated: 2023-09-20
+---
+
+## Time Travel and Rollback
+
+The Iceberg open table format provides a number of benefits to users, including the ability to see a table as it existed at a point in the past. This time travel capability is useful in a number of different ways. For example, having the ability to query historical data is useful for auditing purposes. Or if an application corrupts table data in some way, you can imagine the value in being able to quickly reverse those changes by resetting the table to a known good state.
+
+Iceberg uses snapshots to support this time travel capability. A snapshot represents the state of a table at some point in time. When data is modified in a table, such as inserting, updating, or deleting records, a new snapshot is created. There are maintenance operations that can be used to clean up older snapshots that are no longer needed.
+
+As of 3Q 2023, time travel queries are not yet supported in watsonx.data. However, you can roll back a table to an earlier point in time using snapshots. You will go through that process in this section.
+
+
+1. Select the **Query workspace** icon from the left-side menu.
+
+2. Copy and paste the following SQL statement into the **SQL worksheet**. Click **Run on presto-01**.
+
+  ```bash
+    create table iceberg_data.my_schema.airport_id as select * from hive_data.ontime.airport_id;
+  ```
+  ![](./images/108/result.png)
+
+  As shown in the output above after running the query, there are 6,250 rows in this table.
+
+3. Select the **Data manager** icon from the left-side menu.
+
+4. Navigate to the **iceberg_data** > **my_schema** > **airport_id** table (if you don’t see the table,refresh the schema). Then, select the **Time travel** tab.
```

Comment:
Insert space before "refresh the schema."

Reply:
done